donation/views.py

```diff
@@ -51,16 +74,27 @@ def donation_list(request):
 
     return render(request, 'donation/list.html', context)
 
+
+@login_required(login_url='/admin/login')
+@ong_required
 def donation_update(request, donation_id):
 
     donation = get_object_or_404(Donation, id=donation_id)
-    form = CreateNewDonation(instance = donation)
-
+    if request.user.is_anonymous:
```

Comment:
You should delete this `if` section. If the login is required, then you will never reach to this point as you will never enter this form as anonymous user.